### PR TITLE
Examples' CMake build fix

### PR DESCRIPTION
**Pull request by xzhangxa**
1. Fix bpftool build error, follow commit fc28424eb3f0e39cfb5959296b070389b9a8bd8f to make bootstrap only and set the right path.

```
jit_disasm.c: In function ‘disasm_print_insn’:
jit_disasm.c:102:17: error: too few arguments to function ‘init_disassemble_info’
  102 |                 init_disassemble_info(&info, stdout,
      |                 ^~~~~~~~~~~~~~~~~~~~~
In file included from jit_disasm.c:23:
/usr/include/dis-asm.h:472:13: note: declared here
  472 | extern void init_disassemble_info (struct disassemble_info *dinfo, void *stream,
      |             ^~~~~~~~~~~~~~~~~~~~~
jit_disasm.c:105:17: error: too few arguments to function ‘init_disassemble_info’
  105 |                 init_disassemble_info(&info, stdout,
      |                 ^~~~~~~~~~~~~~~~~~~~~
/usr/include/dis-asm.h:472:13: note: declared here
  472 | extern void init_disassemble_info (struct disassemble_info *dinfo, void *stream,
      |             ^~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:216: /home/debian/projects/libbpf-bootstrap/examples/c/build/bpftool/jit_disasm.o] Error 1
make[2]: *** [CMakeFiles/bpftool.dir/build.make:86: bpftool/src/bpftool-stamp/bpftool-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:159: CMakeFiles/bpftool.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

2. Fix `minimal` bpf object file build error. Quote of `CLANG_SYSTEM_INCLUDES` should be removed for `add_custom_command`.

```
[ 56%] [clang] Building BPF object: minimal
In file included from /home/debian/projects/libbpf-bootstrap/examples/c/minimal.bpf.c:3:
In file included from /usr/include/linux/bpf.h:11:
/usr/include/linux/types.h:5:10: fatal error: 'asm/types.h' file not found
         ^~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/minimal.dir/build.make:78: minimal.bpf.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:288: CMakeFiles/minimal.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

3. Fix `profile` app build which is added at commit 02b8c630faf65888289012edd381d6b2f1197094.

Signed-off-by: Xin Zhang <zhangxin.xa@gmail.com>

### CODE REVIEW

**Comment by anakryiko** [comment]
@xzhangxa can you please split this change into three separate commits, as they are all self-contained fixes? And a bit cleaner commit message with explanation of what they are doing would be much appreciated as well. Thanks!

**Comment by xzhangxa** [comment]
> @xzhangxa can you please split this change into three separate commits, as they are all self-contained fixes? And a bit cleaner commit message with explanation of what they are doing would be much appreciated as well. Thanks!

Done, the changes are split into 3 commits, please check the commit comments, thanks.

For the last one building `bootstrap` only for `bpftool`, I checked again and found bpftool `make` failure shows on Debian 11 and Debian testing, but works fine on Ubuntu 22.04. `make bootstrap` works on 2 Debian machines and it's what the Makefile does. It's like a workaround to the real problem, but since it's aligned with Makefile and enough for the examples so I made the change.

File: examples/c/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +59,9 @@ find_package(BpfObject REQUIRED)
 
 # Create an executable for each application
 file(GLOB apps *.bpf.c)
+if(NOT CARGO_EXISTS)
+  list(REMOVE_ITEM apps ${CMAKE_CURRENT_SOURCE_DIR}/profile.bpf.c)
+endif()
```
Comment (anakryiko) [inline examples/c/CMakeLists.txt:64]: do you think it's possible to have a list of apps that require blazesym and do all these modification for them generically? It's fine for now, but thinking ahead, if we add another blazesym example, Cmake will break again. Just adding an app name to some list seems like the best way to keep Cmake working properly

**Comment by anakryiko** [comment]
Sorry, this dropped off from me radar. Thanks for the fixes!

**Pull request merged**